src/agent/src/image_rpc.rs

```diff
@@ -197,8 +194,58 @@ impl ImageService {
                 return Err(e);
             }
         };
+        self.add_image(String::from(image), String::from(cid)).await;
+        Ok(())
+    }
+
+    /// Pull image when creating container and return the bunld path with rootfs.
+    pub async fn pull_image_in_create_container(&self, image: &str, cid: &str) -> Result<String> {
+        Self::set_proxy_env_vars();
+        if image.contains("pause") {
```

Comment:
It's unsafe to detect pause container to check `pause` prefix in image name.
On the other handle, CRI provides an annotation about container type: "pause" vs "normal". So we should make use of the annotation for safety.
Another TODO:)

Reply:
Sure.

Reply:
We should detect container type by checking `io.kubernetes.cri.container-type` in ImagePullVolume.Metadata.

Reply:
I have already handled it in runtime: 
```
if container_type == "sandbox" {
			image_ref = "pause"
		} else {
			image_ref = container_annotations["io.kubernetes.cri.image-name"]
			if image_ref == "" {
				return nil, fmt.Errorf("Failed to get image name from annotations")
			}
		}
```
I believe it's unnecessary to inform the Image service whether it's dealing with a container or a sandbox.

Reply:
I mean we need a reliable way to detect pause image instead of checking the "pause" prefix in image ref.
Checking "io.kubernetes.cri.container-type" is a reliable way to detect pause image.

Reply:
`image.contains("pause")` is really unreliable.

Reply:
Sure.

Reply:
Done.